site/src/contexts/useProxyLatency.ts

```diff
@@ -25,20 +25,29 @@ const proxyLatenciesReducer = (
   action: ProxyLatencyAction,
 ): Record<string, ProxyLatencyReport> => {
   // Just overwrite any existing latency.
-  return {
-    ...state,
-    [action.proxyID]: action.report,
-  }
+  state[action.proxyID] = action.report
+  return state
 }
 
 export const useProxyLatency = (
   proxies?: RegionsResponse,
-): Record<string, ProxyLatencyReport> => {
+): {
+  // Refetch can be called to refetch the proxy latencies.
+  // Until the new values are loaded, the old values will still be used.
+  refetch: () => void
+  proxyLatencies: Record<string, ProxyLatencyReport>
+} => {
   const [proxyLatencies, dispatchProxyLatencies] = useReducer(
     proxyLatenciesReducer,
     {},
   )
 
+  // This fetchNumber is used to trigger a refetch of the proxy latencies.
+  const [fetchNumber, setFetchNumber] = useState(0)
+  const refetch = (): void => {
+    setFetchNumber(fetchNumber + 1)
```

Comment:
this name is ok, but I would do something like this: 

```
const [latestFetchRequest, setLatestFetchRequest] = useState(new Date().toISOString())

const refetch = () => {
  setLatestFetchRequest(new Date().toISOString())
}
```

I think this is a bit better because we can display the latest time the latency was checked if we want to and the naming is easier to reason about but it is up to you.

Reply:
Each latency report has the date it was checked. I was thinking about adding an integer counter so the FE can tell if the latency provided is the latest one requested. But I figured if we want to add UI flair, we can implement it then.

I will update the name, and make it a date, but won't do anything with the date for now.